core/templates/dev/head/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,130 @@
+      this.urlTemplate, { opportunityType }
+    ), { params }).toPromise().then((data: any) => {
+      const opportunities = [];
+      for (const index in data.opportunities) {
+        opportunities.push(this._getOpportunityFromDict(
+          opportunityType, data.opportunities[index]));
       }
-    };
-    var _fetchOpportunities = function(
-        opportunityType, params, successCallback) {
-      return $http.get(
-        UrlInterpolationService.interpolateUrl(
-          urlTemplate, {opportunityType: opportunityType}
-        ), {
-          params: params
-        }).then(function(response) {
-        var data = response.data;
-        var opportunities = [];
-        for (var index in data.opportunities) {
-          opportunities.push(_getOpportunityFromDict(
-            opportunityType, data.opportunities[index]));
-        }
+      if (successCallback) {
         successCallback(opportunities, data.next_cursor, data.more);
-      });
-    };
-    return {
-      fetchSkillOpportunities: function(cursor, successCallback) {
-        var params = {
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_SKILL, params, successCallback);
-      },
-      fetchTranslationOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_TRANSLATION, params, successCallback);
-      },
-      fetchVoiceoverOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_VOICEOVER, params, successCallback);
       }
+    }, (error) => {
+      if (errorCallback) {
+        errorCallback(error);
+      }
+    });
+  }
+
+  // TODO(#7165): Replace any with exact type.
+  fetchSkillOpportunities(
+      cursor: any
+  ): Promise<object> {
+    const params: ContributionOpportunityParams = {
+      cursor: cursor
+    };
+    return new Promise((resolve, reject) => {
+      this._fetchOpportunities(
+        constants.OPPORTUNITY_TYPE_SKILL, params, resolve, reject);
+    });
+  }
+
+  // TODO(#7165): Replace any with exact type.
+  fetchTranslationOpportunities(
+      languageCode: string,
+      cursor: any
+  ): Promise<object> {
+    const params: ContributionOpportunityParams = {
+      language_code: languageCode,
+      cursor: cursor
     };
-  }]);
+    return new Promise((resolve, reject) => {
+      this._fetchOpportunities(
+        constants.OPPORTUNITY_TYPE_TRANSLATION,
+        params, resolve, reject);
+    });
+  }
+
+  // TODO(#7165): Replace any with exact type.
+  fetchVoiceoverOpportunities(
+      languageCode: string,
+      cursor: any
+  ): Promise<object> {
```

Comment:
Change this to **Promise<Object>**(mind the capital O)

Reply:
Done